### PR TITLE
add certificate check

### PR DESCRIPTION
**Pull request by HaoXuan40404**

### CODE REVIEW

File: tools/check_node_config.sh [hunk 1]
```diff
@@ -48,8 +49,50 @@ print_config_info() {
...
+    if [[ "${nodeid_from_cert}x" != "${nodeid_from_file}x" ]]; then
+        LOG_WARN "nodeid match failed!"
+        LOG_WARN "nodeid from ${node_cert} is ${nodeid_from_cert}"
+        LOG_WARN "nodeid from ${path}/conf/node.nodeid is ${nodeid_from_file}"
+        return 1
+    fi
+    LOG_INFO "match nodeid successful"
+
+    # check if the node.nodeid is in group.x.genesis
+    if grep -w ${nodeid_from_file} ${path}/conf/group.*.genesis &>/dev/null; then
```
Comment (bxq2011hust) [inline tools/check_node_config.sh]: delete this check, because error log can't help user

File: tools/check_node_config.sh [hunk 2]
```diff
@@ -48,8 +49,50 @@ print_config_info() {
 }
 
 check_cert() {
-    local config_file=$1
-    # TODO: verify cert and nodeID
+    local path=$1
+    # use ca_cert verify node_cert
+    ca_name=$(awk -F "=" '/ +ca_cert=/ {print $2}' ${path}/config.ini)
+    node_name=$(awk -F "=" '/ +cert=/ {print $2}' ${path}/config.ini)
+    ca_cert=${path}/conf/${ca_name}
```
Comment (bxq2011hust) [inline tools/check_node_config.sh]: `conf` also need get from config.ini

Comment (HaoXuan40404) [reply tools/check_node_config.sh]: remove it now

File: tools/check_node_config.sh [hunk 3]
```diff
@@ -48,8 +49,46 @@ print_config_info() {
...
+    # check if the node.nodeid match with node.crt
+    nodeid_from_cert=$(openssl x509  -text -in ${node_cert} | sed -n "15,20p" |  sed "s/://g" | tr "\n" " " | sed "s/ //g" | cut -c 3-130)
+    nodeid_from_file=$(cat ${data_path}/node.nodeid)
+    if [[ "${nodeid_from_cert}" != "${nodeid_from_file}" ]]; then
+        LOG_WARN "nodeid match failed!"
+        LOG_WARN "nodeid from ${node_cert} is ${nodeid_from_cert}"
+        LOG_WARN "nodeid from ${data_path}/node.nodeid is ${nodeid_from_file}"
+        return 1
+    fi
+    LOG_INFO "The contents of ${data_path}/node.nodeid and ${node_cert} are consistent."
```
Comment (bxq2011hust) [inline tools/check_node_config.sh]:
```suggestion
    LOG_INFO "The contents of ${data_path}/node.nodeid and ${node_cert} are same."
```

**Pull request merged**